mlflow/spark.py

```diff
@@ -794,6 +792,8 @@ def load_model(model_uri, dfs_tmpdir=None, dst_path=None):
         ):
             return PipelineModel.load(mlflowdbfs_path)
 
+    local_model_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=dst_path)
+    _add_code_from_conf_to_system_path(local_model_path, flavor_conf)
```

Comment:
This change is not related to supporting ADLS Gen 2.
I moved these two lines after checking mlflowdbfs because mlflowdbfs doesn't need `_download_artifact_from_uri` to download the model to the driver's local filesystem. The model will be downloaded by mlflowdbfs. So without this code change, the model will be downloaded twice, which I think is unnecessarily harming the performance.

However, it looks like the `local_model_path` is needed by `_add_code_from_conf_to_system_path`, which should be called before loading the model. Could anyone help me understand in which scenario would Spark model need to use the custom code path? I looked at the test [here](https://src.dev.databricks.com/mlflow/mlflow/-/blob/tests/spark/test_spark_model_export.py?L829)  and I still don't get why we need to add the `__file__` to the system path.

If we do need to support this, we should REVERT the code change in this file.

Reply:
I believe this is the only open question.  cc @jinzhang21 who may have context on this?